### ISSUE TITLE
Composer is crashing after selfupdate

### ISSUE THREAD

**Issue by edhgoose** [composer/composer#4979]
Possibly due to:

https://github.com/composer/composer/commit/1aec1c1fc8b8ac3d2934d819f38d05e54caf552e

And because `default` is a reserved word. This is the specific line:

https://github.com/composer/composer/commit/1aec1c1fc8b8ac3d2934d819f38d05e54caf552e#diff-c0a69bfcfe0dcc5ec2d20757275755d3R78

````
PHP Parse error:  syntax error, unexpected 'default' (T_DEFAULT) in phar:///usr/local/bin/composer/src/Composer/Command/InitCommand.php on line 348
PHP Stack trace:
PHP   1. {main}() /usr/local/bin/composer:0
PHP   2. require() /usr/local/bin/composer:25
PHP   3. Composer\Console\Application->__construct() phar:///usr/local/bin/composer/bin/composer:42
PHP   4. Symfony\Component\Console\Application->__construct() phar:///usr/local/bin/composer/src/Composer/Console/Application.php:85
PHP   5. Composer\Console\Application->getDefaultCommands() phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:90
PHP   6. spl_autoload_call() phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:291
PHP   7. Composer\Autoload\ClassLoader->loadClass() phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:291
PHP   8. Composer\Autoload\includeFile() phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php:301

Parse error: syntax error, unexpected 'default' (T_DEFAULT) in phar:///usr/local/bin/composer/src/Composer/Command/InitCommand.php on line 348

Call Stack:
    0.2028     231024   1. {main}() /usr/local/bin/composer:0
    0.2077     426688   2. require('phar:///usr/local/bin/composer/bin/composer') /usr/local/bin/composer:25
    0.2109     911504   3. Composer\Console\Application->__construct() phar:///usr/local/bin/composer/bin/composer:42
    0.2114     945320   4. Symfony\Component\Console\Application->__construct() phar:///usr/local/bin/composer/src/Composer/Console/Application.php:85
    0.2164    1691744   5. Composer\Console\Application->getDefaultCommands() phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:90
    0.2199    2216624   6. spl_autoload_call() phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:291
    0.2199    2216672   7. Composer\Autoload\ClassLoader->loadClass() phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:291
    0.2199    2216800   8. Composer\Autoload\includeFile() phar:///usr/local/bin/composer/vendor/composer/ClassLoader.php:301
````

**Comment by YetiCGN** [comment]
Same behavior here on PHP 5.5.9. Rollback also not possible!

**Comment by edhgoose** [comment]
Ping @Seldaek

**Comment by elgrafico** [comment]
writing this comment get informed on updates - this is crucial as installing composer via
```
curl -s https://getcomposer.org/installer | php
```
is affected as well.

**Comment by nicolaskern** [comment]
The same just happened in here with PHP 5.5.30, and I'm deploying using capifony so I can't find another solution :(

**Comment by Seldaek** [comment]
Sorry folks, I'll push a fix shortly :/

**Comment by Seldaek** [comment]
Ok should be fixed if you self-update again. Sorry again for the trouble, I didn't remember `default` was a reserved word in php5 :/

**Comment by YetiCGN** [comment]
Thanks @Seldaek for the quick reaction!

**Comment by nicolaskern** [comment]
Indeed, many thanks!
For those being stuck after a selfupdate : curl -s https://getcomposer.org/installer | php && sudo mv composer.phar /usr/local/bin/composer

**Comment by Seldaek** [comment]
I guess you can't self-update if it crashes on init, but in that case please follow install instructions at https://getcomposer.org/download/ to reinstall it.

**Comment by edhgoose** [comment]
Confirmed fixed. Cheers @nicolaskern - I had the same problem.

**Comment by jbrinksmeier** [comment]
confirmed fix.
Also big thanks for the quick fix @Seldaek

**Comment by npotier** [comment]
Thanks @nicolaskern :+1:

**Comment by staabm** [comment]
I wonder why you guys run into this problem just a few mins after the faulty commit was done.

Do you always use `self-update` in a continous process before running composer commands?

**Comment by barryvdh** [comment]
Well, as mentioned, some use Capifony, which self-updates by default on deployments. And a lot of projects run self-update as part of their CI builds. And of course, Composer is installed and updated regularly by thousands of developers on multiple servers etc, so it's not weird that a few people run in to this fast?

**Comment by edhgoose** [comment]
@staabm Blindly is not quite right, but yes. I can't remember exactly why but it was to resolve another problem we had, possibly related to Capifony as @barryvdh mentions.

**Comment by staabm** [comment]
> Well, as mentioned, some use Capifony, which self-updates by default on deployments.

this should be fixed then. doesnt make sense.

> And a lot of projects run self-update as part of their CI builds

this is "ok" when you need bleeding edge composer features, but for most use cases also doesnt have benefit and makes the CI build unstable.

**Comment by staabm** [comment]
just looked up "self-update" in Campifony.. it is possible to disable the composer self-update, see https://github.com/everzet/capifony/issues/413

Having a deployment process which depends on external (and in this case even unstable development builds) is a very bad idea.
I will not argue further, make your own decision.

**Comment by YetiCGN** [comment]
FWIW, I discovered it after attempting a manual `composer update` on a server, because I got all those Symfony Roadmap Notifications and ran into the bug. That was just bad luck.

However, I don't think auto-updating is a bad idea per se. Think security issues that you don't want to track and assess seperately. We just need stable branches, which were simply not available until recently. I think with the 1.0 version coming out, things will calm down and `self-update` should then be pegged to a stable branch (unless you specify otherwise with a command line switch).
